library/x509_crt.c

```diff
@@ -1025,6 +1026,13 @@ static int x509_get_crt_ext( unsigned char **p,
             if( ( ret = x509_get_certificate_policies( p, end_ext_octet,
                     &crt->certificate_policies ) ) != 0 )
             {
+                /* Give the callback (if any) a chance to handle the extension
+                 * if it contains unsupported policies */
+                if( ret == MBEDTLS_ERR_X509_FEATURE_UNAVAILABLE && cb != NULL &&
+                    cb( p_ctx, crt, &extn_oid, is_critical,
+                        start_ext_octet, end_ext_octet ) == 0 )
+                    ret = 0;
```

Comment:
If the certificate policies are validated by `x509_get_certificate_policies()` then we get out of the switch and go back to the while loop. 
If the certificate policies are validated by the callback, we return 0 at line 1038 or line 1048.
Isn't it a problem? Shouldn't we add a `break;` after `ret = 0;`?

Reply:
You are totally right and I fixed it (force pushed it). It was just sufficient to replace `ret = 0;` with `break;` so that the loop is continued when the callback returns 0.